libdnf/goal/IdQueue.hpp

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2017 Red Hat, Inc.
+ *
+ * Licensed under the GNU Lesser General Public License Version 2.1
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+#ifndef __DNF_QUEUE_HPP
+#define __DNF_QUEUE_HPP
+
+extern "C" {
+#include <solv/queue.h>
+}
+
+namespace libdnf {
+
+struct IdQueue {
+public:
+    IdQueue();
+    IdQueue(IdQueue & src);
+    explicit IdQueue(Queue & src);
+    ~IdQueue();
+
+    void pushBack(Id id);
+    void pushBack(Id id1, Id id2);
+    Id operator[](int index) const;
+    int * getElements() const noexcept;
```

Comment:
I prefer renaming to data() (compatible with standard library).
Eg. std::vector has two methods that return pointer to the underlying array serving as element storage:
```
T* data() noexcept;
const T* data() const noexcept;
```